src/zappy_gui_src/Core/SFML/Map/Map.cpp

```diff
@@ -8,3 +8,73 @@
 /// \file src/zappy_gui_src/Core/SFML/Map/Map.cpp
 
 #include "Map.hpp"
+
+using namespace gui;
+
+Map::Map()
+{
+    _tileSelected = -1;
+    _tileHover = -1;
+    _mapSize = {10, 10};
+    _zoom = 1;
+}
+
+void Map::_updateTileVectorSize()
+{
+    while (_mapSize.x * _mapSize.y < _tile.size())
+        _tile.pop_back();
+    while (_mapSize.x * _mapSize.y > _tile.size()) {
+        _tile.push_back(std::make_unique<Tile>());
+        _tile.back()->setPosition(sf::Vector2f((_tile.size() - 1) % (int)_mapSize.x, (_tile.size() - 1) / (int)_mapSize.x));
+        _tile.back()->setSize(sf::Vector2f(100, 100));
```

Comment:
What happen if setTexture will fail

Reply:
if the image can't be load, the program won't crash. The color of the shape is set to white by default, and a message is print on the terminal.

Reply:
but @waroth have a compilation error, so I will check the return value